saleor/graphql/product/mutations/products.py

```diff
@@ -336,8 +336,13 @@ def clean_sku(cls, product_type, cleaned_input, errors):
         See the documentation for `has_variants` field for details:
         http://docs.getsaleor.com/en/latest/architecture/products.html#product-types
         """
-        if not product_type.has_variants and not cleaned_input.get('sku'):
-            cls.add_error(errors, 'sku', 'This field cannot be blank.')
+        if not product_type.has_variants:
+            input_sku = cleaned_input.get('sku')
+            if not input_sku:
+                cls.add_error(errors, 'sku', 'This field cannot be blank.')
+            elif models.ProductVariant.objects.filter(sku=input_sku).exists():
+                cls.add_error(
+                    errors, 'sku', 'Product with this Sku already exists.')
```

Comment:
Nitpick: could you rewrite "Sku" to "SKU"? We rather use the latter spelling. Same in the other change below.

Reply:
Context: SKU stands for "stock-keeping unit" and represents the warehouse code of the item.